src/test/java/org/zalando/nakadi/controller/SubscriptionControllerTest.java

```diff
@@ -176,8 +194,33 @@ public void whenGetNoneExistingSubscriptionThenNotFound() throws Exception {
                 .andExpect(content().string(jsonHelper.matchesObject(expectedProblem)));
     }
 
+    @Test
+    public void whenPostSubscriptionThenWithNoReadScope() throws Exception {
+        when(eventTypeRepository.findByName("myET")).thenReturn(EventTypeTestBuilder.builder()
+                .name("myET")
+                .readScopes(Collections.singleton("oauth.read.scope"))
+                .build());
+
+        expectedException.expectCause(IsInstanceOf.instanceOf(IllegalScopeException.class));
+        expectedException.expectMessage("Client has to have scopes: [oauth.read.scope]");
```

Comment:
In spring-test 4.2 it's very easy to integrate to mockMvc:
http://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/test/web/servlet/setup/StandaloneMockMvcBuilder.html#setControllerAdvice-java.lang.Object...-
You will just add setControllerAdvice(new ExceptionHandling()) when building mockMvc
I think it's a good reason to upgrade to spring-test 4.2 :)